crates/ironrdp-server/src/display.rs

```diff
@@ -56,6 +56,44 @@ pub struct ColorPointer {
     pub xor_mask: Vec<u8>,
 }
 
+pub struct Framebuffer {
+    pub width: NonZeroU16,
+    pub height: NonZeroU16,
+    pub format: PixelFormat,
+    pub data: BytesMut,
```

Comment:
question: What is the benefit of using `BytesMut` at this place?

Reply:
it's not nessarily a Vec<u8>, we may have shared memory. Although I realize shared memory may be problematic too... Perhaps a whole copy is inavoidable..

Reply:
User has to be careful that shared memory will be owned by the server. But we should still allow shared memory to avoid copy.

Reply:
I assume that at some point in the future `freeze()` is called, and memory is actually shared then?
This sounds good to me, but as you point out, let’s be careful about how we use this: we experienced issues in the past where the way BytesMut was used caused too much memory was being allocated and freed, resulting in both performance and memory usage being poor.
I assume you are measuring the CPU and memory usage, and have a good insight into what you are doing.